typescript/sdk/src/core/MultiProtocolCore.ts

```diff
@@ -0,0 +1,71 @@
+import debug from 'debug';
+
+import { ProtocolType } from '@hyperlane-xyz/utils';
+
+import { AdapterClassType, MultiProtocolApp } from '../app/MultiProtocolApp';
+import {
+  HyperlaneEnvironment,
+  hyperlaneEnvironments,
+} from '../consts/environments';
+import { MultiProtocolProvider } from '../providers/MultiProtocolProvider';
+import { TypedTransactionReceipt } from '../providers/ProviderType';
+import { ChainMap, ChainName } from '../types';
+
+import { EvmCoreAdapter } from './adapters/EvmCoreAdapter';
+import { SealevelCoreAdapter } from './adapters/SealevelCoreAdapter';
+import { ICoreAdapter } from './adapters/types';
+import { CoreAddresses } from './contracts';
+
+export class MultiProtocolCore extends MultiProtocolApp<
```

Comment:
Ideally this would have been added in #2529 but I hadn't yet realized we'd need it for this